Login as Customer for 2.4.0 (#563)
magento/merchdocs: src/configuration/customers/login-as-customer.md

---
title: Login as Customer
---

There are times when customers need help with their order. To see what they see, store administrators can "Login as Customer," which allows them to see what the customer sees and take actions on their behalf.

Any actions taken while logged in "as Customer" will be applied to the actual customer's account.

## To enable Login as Customer:

Login as Customer is enabled in the Access Control List.

1. Go to **System** > Permissions > **User Roles**.
1. Select a Role to edit.
1. Select Role Resources > Role Resouces. Change to 'Custom'.

> 1. Under 'Customer', there are options for:

Reviewer: This isn't really a directive, just a statement. 

```suggestion
1. Scroll to the  **Customers**  parent item and the **Login as Customer** item underneath. Then select the resources that you want to enable for the role:
```

I would put in a screen here that shows an example -- a cut out and not the whole tree, just the ones you want them to know about it.

>    *  Login as Customer

Reviewer: Remove this and move the other bullets up a level.

Author: Done

>       *  Allow Login as Customer - This allows the admin user to use the Login as Customer feature.

Reviewer: ```suggestion
      *  **Allow Login as Customer** - This allows the Admin user to use the _Login as Customer_ feature.
```

>       *  View Login as Customer Log - Allows the admin user to see the Login as Customer Log.

Reviewer: ```suggestion
      *  **View Login as Customer Log** - Allows the Admin user to see the _Login as Customer_ Log.
```

> 

Reviewer: I think you need a save step here:

```suggestion
1. Click <span class="btn">Save Role</span>.
```

> Then, go to **Stores** > **Configuration** > **Customers** > **Login as Customer**.

Reviewer: This is more than one menu/UI area and needs to be broken up. 

```suggestion
1. On the _Admin_ sidebar go to  **Stores** > _Settings_ > **Configuration**.

1. In the left panel, expand **Customers** and choose  **Login as Customer**.
```

> 

Reviewer: This looks like it needs to be a new page in the config reference.

Author: Done

| Option | Values | Description |
|-- | -- | -- |
| Enable Login as Customer | Yes/No | Enable/Disable the Login as Customer feature |
| Disable Page Cache for Admin User | Yes/no | If Yes, the Page Cache will be disabled for the admin user. |
| Store View to Log in | Automatic/Manual | Automatic will use the default store setting for that company user. Use "Manual Choose" if want the admin user to select the store view when logging in as Customer. |

Disabling the page cache will ensure that the user logging in as Customer will get fresh, uncached data.

Once enabled, the Login as Customer button will appear in the:

*  [Customer Edit page](https://docs.magento.com/m2/ce/user_guide/customers/customer-account-update.html)
*  [Order View page](https://docs.magento.com/m2/ce/user_guide/sales/order-processing.html#view-an-order)
*  [Invoice View page](https://docs.magento.com/m2/ce/user_guide/sales/invoice-create.html)
*  [Shipment View page](https://docs.magento.com/m2/ce/user_guide/sales/shipments-create.html)
*  [Credit Memo View page](https://docs.magento.com/m2/ce/user_guide/sales/credit-memo-create.html)

> ## Login as Customer Log

Reviewer: ```suggestion
## Login as Customer log
```

> Magento provides a log for the Login as Customer feature. It lists all sessions where a store administrator used the Login as Customer feature. It can be found at XXXXXXXX.

Reviewer: What is XXXXXX?

```suggestion
Magento provides a log for the _Login as Customer_ feature. It lists all sessions where an Admin user accesses the feature. It can be found at XXXXXXXX.
```

> <!-- Enterprise Edition -->

Reviewer: This needs an ee tag? If the log is a Commerce-only feature, put the `{:.ee-only}` tag directly above the Login as Customer Log heading.

Author: Done

> Magento provides a log for Login as Customer actions. In Magento Commerce, actions taken under Login as Customer will be listed in the main Magento log.

Reviewer: ```suggestion
Magento provides a log for _Login as Customer_ actions. In Magento Commerce, actions taken under Login as Customer are listed in the main Magento log.
```

<!-- /Enterprise Edition -->

> ## Permissions

Reviewer: This section seems redundant. You already outlined the permissions above. I would suggest adding sentence or two in the following "Using Login as Customer" section that the permissions described in the earlier section are required to access these features. I would not call them "Access Control options" .

Reviewer: I split as two topics - concept procedure in main customers section and config ref.

> The Login as Customer feature is enabled for Roles in the Access Control List.

Reviewer: "Access Control List"

```suggestion
The Login as Customer feature is enabled for user roles in the Role Resources configuration.
```

> Role Resources are set in **System** > Permissions > **User Roles**.

Reviewer: ```suggestion
To set the Role Resources , go to **System** > _Permissions_ > **User Roles**.
```

> Access Control can be set when you create a new user or edit an existing one.

Reviewer: ```suggestion
You can control user access when you create a new user or edit an existing one.
```

> There are two Access Control options:

Reviewer: I would get rid of this.

*  Login as Customer Button
*  Login as Customer Log

Set these for each Role you wish to have this feature.

## Using Login as Customer

> Login as Customer allows you to see the site just as the customer does, and allows you to troubleshoot and take other actions for the customer. The feature is expressed a 'Login as Customer' button in the locations listed above, and the Login as Customer Log will be available in the Admin.

Reviewer: ```suggestion
Login as Customer allows you to see the site just as the customer does, and allows you to troubleshoot and take other actions for the customer. If you have an assigned user role with the required permissions:

* You can click <span class="btn">Login as Customer</span> on the pages listed in the previous section.
* The Login as Customer Log is available in the Admin.
```

{.bs-callout-warning}

> Any actions taken while logged in "as Customer" (add/remove products, etc.) will be applied to the actual customer's order.

Reviewer: ```suggestion
Any actions taken while logged in "as Customer" (add/remove products, etc.) are applied to the actual customer's order.
```

> On the storefront, as banner will appear when logged in as Customer, to provide an indication of the special state.

Reviewer: ```suggestion
On the storefront, a banner is displayed when you are "logged in as customer" to provide a reminder of the special state.
```
